src/lib.rs

```diff
@@ -70,184 +54,43 @@ pub enum Backlight {
-            DisplayControl::DisplayOn as u8
-        };
-        let display_ctrl = if self.cursor_blink {
-            display_ctrl | DisplayControl::CursorBlink as u8
-        } else {
-            display_ctrl
-        };
-        self.command(Mode::DisplayControl as u8 | display_ctrl)?;
-        self.command(Mode::Cmd as u8 | Commands::Clear as u8)?; // Clear Display
-
-        // Entry right: shifting cursor moves to right
-        self.command(0x04)?;
-        self.backlight(self.backlight_state)?;
-        Ok(self)
-    }
-
-    fn write4bits(&mut self, data: u8) -> Result<(), I::Error> {
-        self.i2c.write(
-            self.address,
-            &[data | DisplayControl::DisplayOn as u8 | self.backlight_state as u8],
-        )?;
-        self.i2c.write(
-            self.address,
-            &[DisplayControl::Off as u8 | self.backlight_state as u8],
-        )?;
-        self.delay.delay_us(700);
-        Ok(())
-    }
-
-    fn send(&mut self, data: u8, mode: Mode) -> Result<(), I::Error> {
-        let high_bits: u8 = data & 0xf0;
-        let low_bits: u8 = (data << 4) & 0xf0;
-        self.write4bits(high_bits | mode as u8)?;
-        self.write4bits(low_bits | mode as u8)?;
-        Ok(())
-    }
-
-    fn command(&mut self, data: u8) -> Result<(), I::Error> {
-        self.send(data, Mode::Cmd)
-    }
-
-    pub fn backlight(&mut self, backlight: Backlight) -> Result<(), I::Error> {
-        self.backlight_state = backlight;
-        self.i2c.write(
-            self.address,
-            &[DisplayControl::DisplayOn as u8 | backlight as u8],
-        )
-    }
-
-    /// Write string to display.
-    pub fn write_str(&mut self, data: &str) -> Result<(), I::Error> {
-        for c in data.chars() {
-            self.send(c as u8, Mode::Data)?;
-        }
-        Ok(())
-    }
-
-    /// Clear the display
-    pub fn clear(&mut self) -> Result<(), I::Error> {
-        self.command(Commands::Clear as u8)?;
-        self.delay.delay_ms(2);
-        Ok(())
-    }
-
-    /// Return cursor to upper left corner, i.e. (0,0).
-    pub fn return_home(&mut self) -> Result<(), I::Error> {
-        self.command(Commands::ReturnHome as u8)?;
-        self.delay.delay_ms(2);
-        Ok(())
-    }
-
-    /// Set the cursor to (rows, col). Coordinates are zero-based.
-    pub fn set_cursor(&mut self, row: u8, col: u8) -> Result<(), I::Error> {
-        self.return_home()?;
-        let shift: u8 = row * 40 + col;
-        for _i in 0..shift {
-            self.command(Commands::ShiftCursor as u8)?;
-        }
-        Ok(())
-    }
+#[repr(u8)]
+#[derive(Copy, Clone)]
+pub enum Font {
+    Font5x8 = 0x00,
+    Font5x10 = 0x04, 
 }
 
-impl<'a, I, D> uWrite for Lcd<'a, I, D>
-where
-    I: I2c,
-    D: DelayNs,
-{
-    type Error = I::Error;
-
-    fn write_str(&mut self, s: &str) -> Result<(), Self::Error> {
-        self.write_str(s)
-    }
+#[repr(u8)]
+#[derive(Copy, Clone)]
+enum CursorMoveDir {
+    Right = 0x00,
+    Left = 0x02,
```

Comment:
According to [the lib](https://github.com/marcmerlin/NewLiquidCrystal/blob/a752411879b8aae0c6846d24ab3abb9d73694e79/LCD.h#L83), `Left` should be `0x00` and `Right` is `0x04`, right?

Reply:
I think I confused the constants here, `0x02` is correct